### PR TITLE
Make DesktopConfigTest and update DesktopConfig

### LINKED ISSUE #2885: Test DesktopConfig.js
DesktopConfig.js is not tested and as is not testable either. The class should be split into the config itself and the persistence mechanisms it uses and after that tested with unit tests.

### PR DESCRIPTION
**Pull request by wrdhub**
close #2885

### CODE REVIEW

File: test/tests/desktop/config/DesktopConfigTest.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
+import o from "ospec"
+import { DesktopConfig } from "../../../../src/desktop/config/DesktopConfig.js"
+import { DesktopConfigMigrator } from "../../../../src/desktop/config/migrations/DesktopConfigMigrator.js"
+import { downcast, noOp } from "@tutao/tutanota-utils"
+import { DesktopKeyStoreFacade } from "../../../../src/desktop/KeyStoreFacadeImpl.js"
+import { DesktopNativeCryptoFacade } from "../../../../src/desktop/DesktopNativeCryptoFacade.js"
+import { DesktopConfigKey } from "../../../../src/desktop/config/ConfigKeys.js"
+import { ConfigFile } from "../../../../src/desktop/config/ConfigFile.js"
+
+o.spec("DesktopConifgTest", function () {
```
Comment (ganthern) [inline test/tests/desktop/config/DesktopConfigTest.ts:10]: typo

File: src/desktop/config/ConfigFile.ts [hunk 2]
```diff
@@ -9,34 +10,37 @@ type FsExports = typeof FsModule
...
 	/**
 	 * @param p path to the file the json objects should be stored in
 	 * @param fs
 	 **/
-	constructor(p: string, fs: FsExports) {
-		this._path = p
-		this._accessPromise = Promise.resolve()
-		this._fs = fs
+	constructor(p: string, f: string, fs: FsExports) {
+		this.path = p
```
Comment (ganthern) [inline src/desktop/config/ConfigFile.ts:35]: path, fullpath and fs could be declared as `private readonly` inside of the constructor signature declaration like you did in `DesktopConfig`

**Review by ganthern — COMMENTED** [review]
is there a reason that you went the manual stubbing route instead of using testdouble for the new test file?

**Comment by wrdhub** [comment]
> is there a reason that you went the manual stubbing route instead of using testdouble for the new test file?

I guess it made the most sense at the time? I think I kind of forgot testdouble existed.

File: src/desktop/config/ConfigFile.ts [hunk 3]
```diff
@@ -9,34 +10,31 @@ type FsExports = typeof FsModule
...
-	 * @param fs
+	 * @param filePath path to the file the json objects should be stored in
 	 **/
-	constructor(p: string, fs: FsExports) {
-		this._path = p
-		this._accessPromise = Promise.resolve()
-		this._fs = fs
+	constructor(private readonly filePath: string, private readonly fullpath: string, private readonly fs: FsExports) {
+		this.accessPromise = Promise.resolve()
+		this.fs = fs
```
Comment (ganthern) [inline src/desktop/config/ConfigFile.ts:32]: that line can be removed now I think

**Review by ganthern — APPROVED** [review]

**Pull request merged**